File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:
    bool isPalindrome(string s){
        int L = 0;
        int R = s.length()-1;
        while(L < R){
            if(s[L] != s[R]){
                return false;
            }
            L++;
            R--;
            
        }
        return true;
    }
    bool validPalindrome(string s) {
        int L = 0;
        int R = s.length()-1;
        while(L < R){
            string ss = s;
            if(s[L] != s[R]){
                return isPalindrome(ss.erase(L,1)) || isPalindrome(s.erase(R,1));
            }
            L++;
            R--;
        }
        return true;
    }
};

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.py
class Solution:
    def validPalindrome(self, s: str) -> bool:

        delete = False

        l = 0
        r = len(s)-1

        while l < r:
            if s[l] != s[r]:
                skipL , skipR = s[l+1:r+1], s[l:r]
                return skipL == skipL[::-1] or skipR == skipR[::-1]
            l += 1
            r -= 1
        return True
        